sklearn/utils/estimator_checks.py

```diff
@@ -1346,6 +1350,85 @@ def check_estimators_partial_fit_n_features(name, estimator_orig):
         estimator.partial_fit(X[:, :-1], y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_classifier_multioutput(name, estimator):
+    n_samples = 42
+    n_labels = 5
+    tags = _safe_tags(estimator)
+    estimator = clone(estimator)
+    X, y = make_multilabel_classification(random_state=42,
+                                          n_samples=n_samples,
+                                          n_labels=n_labels)
+    estimator.fit(X, y)
+    y_pred = estimator.predict(X)
+
+    assert y_pred.shape == (n_samples, n_labels)
+    assert y_pred.dtype.kind == 'i'
+
+    if hasattr(estimator, "decision_function"):
+        decision = estimator.decision_function(X)
+        assert isinstance(decision, np.ndarray)
+        assert decision.shape == (n_samples, n_labels)
+
+        dec_pred = (decision > 0).astype(np.int)
+        dec_exp = estimator.classes_[dec_pred]
+        assert_array_equal(dec_exp, y_pred)
+
+    if hasattr(estimator, "predict_proba"):
+        y_prob = estimator.predict_proba(X)
+
+        if isinstance(y_prob, list) and not tags['poor_score']:
+            for i in range(n_labels):
+                y_prob[i].shape == (n_samples, n_labels)
+                assert_array_equal(np.argmax(y_prob[i], axis=1)
+                                   .astype(np.int), y_pred[:, i])
+        elif not tags['poor_score']:
+            y_prob.shape == (n_samples, n_labels)
+            assert_array_equal(y_prob.round().astype(int), y_pred)
+
+    if (hasattr(estimator, "decision_function") and
+            hasattr(estimator, "predict_proba")):
+        for i in range(n_labels):
+            a = estimator.predict_proba(X)[:, i]
+            b = estimator.decision_function(X)
+            assert_array_equal(rankdata(a), rankdata(b[:, i]))
+
+    scores = jaccard_score(y_pred, y, average=None)
+    assert scores.dtype == np.dtype('float')
```

Comment:
This is not testing the estimator, but the metric

Reply:
Removed.